### PR TITLE
Add partial data badges

### PR DESCRIPTION
**Pull request by nfmohit**
## Summary

Addresses issue:

- #8142

## Relevant technical choices

This PR adds the partial data badges in Audience Segmentation widget.

### Deviation from IB

- The tooltips appear when the info icon is hovered, instead of the entire badge.
- The IB proposed two separate components for desktop and mobile viewports. This PR implements one and changes appearance based on the viewport.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTile/AudienceTilePagesMetric.js [hunk 1]
```diff
@@ -17,34 +17,45 @@
...
+import useViewOnly from '../../../../../../hooks/useViewOnly';
+
+const { useSelect } = Data;
 
 export default function AudienceTilePagesMetric( {
 	TileIcon,
 	title,
 	topContent,
 	topContentTitles,
+	isTopContentPartialData = false,
```
[the remark this answers is not in the document]
Comment (nfmohit) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTile/AudienceTilePagesMetric.js]: Updated, thank you!

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTile/AudienceTilePagesMetric.js [hunk 2]
```diff
@@ -98,6 +113,14 @@ export default function AudienceTilePagesMetric( {
 			<div className="googlesitekit-audience-segmentation-tile-metric__container">
 				<div className="googlesitekit-audience-segmentation-tile-metric__title">
 					{ title }
+					{ ! isMobileBreakpoint && isTopContentPartialData && (
+						<PartialDataBadge
```
[the remark this answers is not in the document]
Comment (nfmohit) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTile/AudienceTilePagesMetric.js:118]: I agree, updated, please let me know if this looks good now, thanks!

**Comment by nfmohit** [comment]
> Thanks @nfmohit, nice work here. Just making another case for separate components for mobile and desktop, considering their different appearance (Badge vs Notice). Let me know what you think.

I agree, updated, thank you!

**Review by kuasha420 — CHANGES_REQUESTED** [review]
Excellent work @nfmohit! Thanks for being patient with me and addressing my feedback. I've found one small tweak after the refactor, I'll push the fix myself and get this merged. Cheers!

**Review by kuasha420 — APPROVED** [review]
Thanks @nfmohit :tada:

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/PartialDataBadge.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * Internal dependencies
+ */
+import InfoTooltip from '../../../../../components/InfoTooltip';
+
+export default function PartialDataBadge( { tooltipTitle } ) {
+	return (
+		<span className="googlesitekit-audience-segmentation-partial-data-badge">
+			<Fragment>
```
[the remark this answers is not in the document]
Comment (nfmohit) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/PartialDataBadge.js]: Oops, looks like I mistakenly overlooked it, thank you for removing! 🦅

**Pull request merged**